### PR TITLE
jovian-greeter: Some fixes, including brokenness fix

### PR DESCRIPTION
**Pull request by samueldr**
Look at the commits for the rationales.

Big TLDR: some session files could break the greeter.

Also, looking at the log is not great... I left some logging I ended-up needing to figure out an issue.

### CODE REVIEW

File: pkgs/jovian-greeter/greeter.py [hunk 1]
```diff
@@ -91,8 +92,11 @@ def start_session(self, command: List[str], environment: List[str]):
             logging.debug("Failed to stop Plymouth", exc_info=ex)
 
         # greetd before 0.9.0 doesn't support env
```
Comment (samueldr) [inline pkgs/jovian-greeter/greeter.py:94]: I'm not putting myself on the hook for learning python enough *and* greetd to fix this, if it is fixed. Though that would be an appropriate fix, too, maybe?

Comment (K900) [reply pkgs/jovian-greeter/greeter.py:94]: We should just yeet this tbh, even 24.05 (!) is on a new enough greetd to support doing things the normal way.

File: pkgs/jovian-greeter/greeter.py [hunk 2]
```diff
@@ -181,8 +185,17 @@ def _find_sessions(self, sessions: Iterable[str]) -> Optional[Session]:
 
         return None
 
+def handle_exception(exc_type, exc_value, exc_traceback):
```
Comment (K900) [inline pkgs/jovian-greeter/greeter.py:188]: Don't those get logged to greetd journal? I thought they did? Also, should probably exit(1) after.

Comment (samueldr) [reply pkgs/jovian-greeter/greeter.py:188]: They did not when I had the issues where I needed to escape that semicolon...

But I might have been holding it wrong? Or looking at it wrong?

(And yes, probably should exit...)

File: pkgs/jovian-greeter/greeter.py (hunk 1 above)
Comment (samueldr) [reply pkgs/jovian-greeter/greeter.py:94]: Feel free to handle it. (And yes.)

**Comment by K900** [comment]
Merged most of it manually.

**Pull request closed without merging**